### PR TITLE
Fixing api proxy

### PR DESCRIPTION
**Pull request by huguesBouvier**
This PR fixes a bug where the API proxy could not connect to edgeHub through MQTT.
It fixes a bug where only the first certificate from the trustbundle would be used.

Test:
API proxy logs are now clean of ssl error.
MQTT connects correct: Successfully changed the config through the twin.

### CODE REVIEW

File: edge-modules/api-proxy-module/rust-sdk/azure-iot-mqtt/src/io.rs [hunk 1]
```diff
@@ -166,8 +166,32 @@ impl mqtt3::IoSource for IoSource {
...
+					if line == "-----END CERTIFICATE-----" {
+						tls_connector_builder.add_root_certificate(std::mem::take(&mut current_cert));
+					}
+					else if line == "-----BEGIN CERTIFICATE-----" {
+						if !current_cert.is_empty() {
+							return std::io::Error::new(std::io::ErrorKind::Other, "malformed PEM: BEGIN CERTIFICATE without prior END CERTIFICATE");
+						}
+					}
+					else {
+						tls_connector_builder.add_root_certificate(line);
```
Comment (arsing) [inline edge-modules/api-proxy-module/rust-sdk/azure-iot-mqtt/src/io.rs:187]: This should be `current_cert.push_str(line);`

**Review by arsing — DISMISSED** [review]

**Review by arsing — APPROVED** [review]

File: edge-modules/api-proxy-module/rust-sdk/azure-iot-mqtt/src/iotedge_client.rs [hunk 2]
```diff
@@ -75,11 +75,42 @@ impl Client {
...
+					current_cert.push_str("\n-----END CERTIFICATE-----");
+					let current_cert = std::mem::take(&mut current_cert);
+					let certificate =
+						native_tls::Certificate::from_pem(current_cert.as_bytes())
+						.map_err(|err| Error::GetServerRootCertificate(ApiErrorReason::ParseResponseBody(Box::new(err))))?;
+
+					server_root_certificate.push(certificate);
+				}
+				else if line == "-----BEGIN CERTIFICATE-----" {
+					if !current_cert.is_empty() {
```
Comment (varunpuranik) [inline edge-modules/api-proxy-module/rust-sdk/azure-iot-mqtt/src/iotedge_client.rs:100]: Where does current_cert get reset after the 1st cert is processed?

File: edge-modules/api-proxy-module/rust-sdk/azure-iot-mqtt/src/iotedge_client.rs [hunk 3]
```diff
@@ -75,11 +75,42 @@ impl Client {
...
 
-			let certificate =
-				native_tls::Certificate::from_pem(certificate.as_bytes())
-				.map_err(|err| Error::GetServerRootCertificate(ApiErrorReason::ParseResponseBody(Box::new(err))))?;
+			let mut server_root_certificate = vec![];
+
+			let mut current_cert = String::new();
+			let mut lines = certificate.lines();
+	
+			if lines.next() != Some("-----BEGIN CERTIFICATE-----") {
```
Comment (varunpuranik) [inline edge-modules/api-proxy-module/rust-sdk/azure-iot-mqtt/src/iotedge_client.rs:83]: Can you refactor this into a method that you can then unit test?

File: edge-modules/api-proxy-module/rust-sdk/azure-iot-mqtt/src/iotedge_client.rs (hunk 2 above)
Comment (arsing) [reply edge-modules/api-proxy-module/rust-sdk/azure-iot-mqtt/src/iotedge_client.rs:100]: Line 92

**Review by varunpuranik — APPROVED** [review]

File: edge-modules/api-proxy-module/rust-sdk/azure-iot-mqtt/src/iotedge_client.rs (hunk 3 above)
Comment (huguesBouvier) [reply edge-modules/api-proxy-module/rust-sdk/azure-iot-mqtt/src/iotedge_client.rs:83]: that's a good idea. Making a note of it

**Pull request merged**